BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
```

Comment:
This service can be improved following SOLID principles. 
- Code to an `IExchangeRateService` interface
- Instead of instantiating the service manually, use dependency injection (DI) to inject the service for use in other areas of the application
- The interface will also enable us to mock the service for unit testing

See  [Dependency injection in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-8.0)  for more

Example interface
```csharp
public interface IExchangeRateService
{
    Task<IEnumerable<ExchangeRate>> GetDailyRatesAsync(string? sourceCurrency, string[] targetCurrencies, CancellationToken cancellationToken = default);
}
```